### PR TITLE
removing build tags

### PR DESCRIPTION
**Pull request by otherview**
Removes build tags and uses env vars to run specific tests

### CODE REVIEW

File: integration/simulation/simulation_docker_test.go [hunk 1]
```diff
@@ -25,19 +23,25 @@ import (
...
 	enclaveDockerPort = "11000/tcp"
+	dockerTestEnv     = "DOCKER_TEST_ENABLED"
 )
 
 // This test creates a network of L2 nodes, then injects transactions, and finally checks the resulting output blockchain
 // The L2 nodes communicate with each other via sockets, and with their enclave servers via RPC.
 // All nodes live in the same process, the enclaves run in individual Docker containers, and the Ethereum nodes are mocked out.
 // $> docker rm $(docker stop $(docker ps -a -q --filter ancestor=obscuro_enclave --format="{{.ID}}") will stop and remove all images
 func TestDockerNodesMonteCarloSimulation(t *testing.T) {
+	fooAddr := os.Getenv(dockerTestEnv)
```
Comment (joeldudley) [inline integration/simulation/simulation_docker_test.go:40]: "fooAddr"?

File: .github/workflows/build-push-docker.yml [hunk 2]
```diff
@@ -27,4 +27,4 @@ jobs:
         run:  docker build -t obscuro_enclave -f dockerfiles/enclave_local_build.Dockerfile .
 
       - name: Test
-        run: go test -v -tags docker -run TestDocker ./...
+        run: DOCKER_TEST_ENABLED=true go test -v -run TestDocker ./...
```
Comment (joeldudley) [inline .github/workflows/build-push-docker.yml:30]: Isn't it better practice to use an `env:` block here?

File: integration/simulation/simulation_docker_test.go (hunk 1 above)
Comment (otherview) [reply integration/simulation/simulation_docker_test.go:40]: changing it to something more meaningful :)

File: .github/workflows/build-push-docker.yml (hunk 2 above)
Comment (otherview) [reply .github/workflows/build-push-docker.yml:30]: Think for one var I wouldn't bother, but I can change that 👍

**Review by tudor-malene — COMMENTED** [review]
needs some readme instructions on how to run these skipped tests locally.
Otherwise looks good to me if it works

**Review by joeldudley — APPROVED** [review]
LGTM

**Pull request merged**